packages/block-editor/src/components/inserter/style.scss

```diff
@@ -99,6 +117,17 @@ $block-inserter-search-height: 38px;
 	}
 }
 
+// This extra div is needed because
+// flex grow and overflow auto doesn't work well together.
+.block-editor-inserter__scrollable {
```

Comment:
I spent hours trying to make scrolling behave properly without this extra div without any luck. There's a mix between flex, auto-height popover computation, and scrolling that makes it hard to achieve. This solution is decent I think.